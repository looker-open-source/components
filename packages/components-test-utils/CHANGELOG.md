# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.10.4](https://github.com/looker-open-source/components/compare/v0.10.3...v0.10.4) (2021-01-25)

**Note:** Version bump only for package @looker/components-test-utils





## [0.10.3](https://github.com/looker-open-source/components/compare/v0.10.2...v0.10.3) (2021-01-25)

**Note:** Version bump only for package @looker/components-test-utils





## [0.10.2](https://github.com/looker-open-source/components/compare/v0.10.1...v0.10.2) (2021-01-22)


### Bug Fixes

* On build properly set package.json "files" key ([#1806](https://github.com/looker-open-source/components/issues/1806)) ([c81c4bb](https://github.com/looker-open-source/components/commit/c81c4bb625c58ede49957aad90ac8d9d7b2c4b79))





## [0.10.1](https://github.com/looker-open-source/components/compare/v0.10.0...v0.10.1) (2021-01-22)


### Bug Fixes

* Properly build ems/cjs/ts for easy deep-imports ([#1804](https://github.com/looker-open-source/components/issues/1804)) ([2a523cd](https://github.com/looker-open-source/components/commit/2a523cd70b079944376c6dc87c18202e05a97b01))





# [0.10.0](https://github.com/looker-open-source/components/compare/v0.9.29...v0.10.0) (2021-01-19)


### Bug Fixes

* Properly configure build for tree-shaking support ([9202b72](https://github.com/looker-open-source/components/commit/9202b72bc5fb99eb3e1af7d9e56f3dc15b1df2a4))





## [0.9.15]

### Changed

- `@looker/components-test-utils` helpers will no longer produce `globalStyles`
