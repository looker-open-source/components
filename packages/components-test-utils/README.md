# Test Utilities

Shared functions used for mocking and testing Looker UI Components.
