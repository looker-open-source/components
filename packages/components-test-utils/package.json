{
  "name": "@looker/components-test-utils",
  "license": "MIT",
  "version": "0.9.12",
  "main": "lib/index.js",
  "types": "lib/index.d.ts",
  "files": [
    "lib"
  ],
  "repository": {
    "type": "git",
    "url": "https://github.com/looker-open-source/components",
    "directory": "packages/components-test-utils"
  },
  "publishConfig": {
    "access": "public"
  },
  "devDependencies": {
    "@looker/components-providers": "^0.9.12",
    "@testing-library/react": "^10.4.8",
    "@types/enzyme": "^3.10.5",
    "@types/jest": "^26.0.10",
    "@types/react": "^16.9.49",
    "@types/react-test-renderer": "^16.9.3",
    "enzyme": "^3.11.0",
    "jest-styled-components": "^6.3.4",
    "react": "^16.13.1",
    "react-test-renderer": "^16.13.1",
    "styled-components": "^4.4.1"
  },
  "peerDependencies": {
    "react": "^16.11"
  },
  "gitHead": "62febc162e3e45124f403e2c84fc741f68fe6714"
}
