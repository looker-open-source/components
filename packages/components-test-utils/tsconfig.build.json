{
  "extends": "../../tsconfig.build.json",
  "compilerOptions": {
    "rootDir": "./src",
    "outDir": "./lib"
  },
  "include": ["../../types", "src/**/*"]
}
