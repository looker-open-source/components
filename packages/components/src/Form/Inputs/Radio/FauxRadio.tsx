/*

 MIT License

 Copyright (c) 2020 Looker Data Sciences, Inc.

 Permission is hereby granted, free of charge, to any person obtaining a copy
 of this software and associated documentation files (the "Software"), to deal
 in the Software without restriction, including without limitation the rights
 to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 copies of the Software, and to permit persons to whom the Software is
 furnished to do so, subject to the following conditions:

 The above copyright notice and this permission notice shall be included in all
 copies or substantial portions of the Software.

 THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 SOFTWARE.

 */

import styled from 'styled-components'
import { reset } from '@looker/design-tokens'

const dotSize = 6

export const FauxRadio = styled.div`
  ${reset}
  background-color: currentColor;
  border: solid 1px ${({ theme }) => `${theme.colors.ui2}`};
  border-color: currentColor;
  border-radius: 50%;
  color: transparent;
  height: 100%;
  position: relative;
  transition: background-color 25ms linear, border-color 25ms linear,
    box-shadow 25ms linear;
  width: 100%;

  &::after {
    background: #fff;
    border-radius: 50%;
    content: '';
    height: ${dotSize}px;
    position: absolute;
    right: calc(50% - ${dotSize / 2}px);
    top: calc(50% - ${dotSize / 2}px);
    width: ${dotSize}px;
  }
`
