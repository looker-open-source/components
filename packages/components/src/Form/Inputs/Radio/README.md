# Radio

The `Radio` component renders a single radio button on the page, with no accompanying label.
If you are building a form, you probably want to use [`FieldRadioGroup`](/?path=/docs/docs-form-fields-fieldradiogroup--docs) instead as it provides label and group support.

## Checked Property

The `Radio` component accepts the `checked` prop, which sets the radio button as selected.

```tsx
function Checked() {
  return (
    <Space>
      <Radio />
      <Radio checked />
    </Space>
  );
}
```

## Disabled Property

Use the disable property to gray out the radio button, making it un-clickable.

```tsx
function Disabled() {
  return (
    <Space>
      <Radio disabled />
      <Radio disabled checked />
    </Space>
  );
}
```
