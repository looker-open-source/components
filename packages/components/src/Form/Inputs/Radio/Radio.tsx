/*

 MIT License

 Copyright (c) 2020 Looker Data Sciences, Inc.

 Permission is hereby granted, free of charge, to any person obtaining a copy
 of this software and associated documentation files (the "Software"), to deal
 in the Software without restriction, including without limitation the rights
 to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 copies of the Software, and to permit persons to whom the Software is
 furnished to do so, subject to the following conditions:

 The above copyright notice and this permission notice shall be included in all
 copies or substantial portions of the Software.

 THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 SOFTWARE.

 */

import pick from 'lodash/pick'
import React, { forwardRef, Ref } from 'react'
import styled from 'styled-components'
import { reset, space, SpaceProps } from '@looker/design-tokens'
import { InputProps, inputPropKeys } from '../InputProps'
import { FauxRadio } from './FauxRadio'

export interface RadioProps
  extends SpaceProps,
    Omit<InputProps, 'readonly' | 'type' | 'checked' | 'onClick'> {
  checked?: boolean
  className?: string
}

const RadioLayout = forwardRef(
  (props: RadioProps, ref: Ref<HTMLInputElement>) => {
    const { className, ...restProps } = props

    return (
      <div className={className}>
        <input type="radio" {...pick(restProps, inputPropKeys)} ref={ref} />
        <FauxRadio />
      </div>
    )
  }
)

RadioLayout.displayName = 'RadioLayout'

export const Radio = styled(RadioLayout)`
  ${reset}
  ${space}

  height: 1rem;
  position: relative;
  width: 1rem;
  input {
    background: ${(props) => props.theme.colors.field};
    cursor: ${({ readOnly, disabled }) =>
      readOnly || disabled ? 'not-allowed' : undefined};
    height: 100%;
    opacity: 0;
    position: absolute;
    width: 100%;
    z-index: 1;
  }

  input:checked + ${FauxRadio} {
    color: ${({ theme }) => theme.colors.key};
  }

  input:not(:checked) + ${FauxRadio} {
    border-color: ${({ theme }) => theme.colors.ui2};
    background: ${({ theme }) => theme.colors.field};
  }

  input:focus + ${FauxRadio} {
    border-color: ${({ theme }) => theme.colors.keyFocus};
    box-shadow: 0 0 0 2px ${({ theme }) => theme.colors.keyAccent};
    outline: none;
  }

  input:disabled + ${FauxRadio} {
    color: ${({ theme }) => theme.colors.text5};
  }

  input:disabled:not(:checked) + ${FauxRadio} {
    background: transparent;

    &::after {
      background: transparent;
    }
  }
`
