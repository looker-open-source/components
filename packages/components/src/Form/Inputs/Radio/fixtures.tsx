/**
 * Copyright (c) 2023 Google LLC
 * SPDX-License-Identifier: MIT
 */
import React from 'react';
import { Radio } from '../Radio';
import type { RadioProps } from '../Radio';

export function Basic(props: RadioProps) {
  return <Radio {...props} />;
}

export function Disabled() {
  return (
    <>
      <Radio disabled />
      <Radio disabled checked />
    </>
  );
}
