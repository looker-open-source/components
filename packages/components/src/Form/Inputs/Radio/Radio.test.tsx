/*

 MIT License

 Copyright (c) 2020 Looker Data Sciences, Inc.

 Permission is hereby granted, free of charge, to any person obtaining a copy
 of this software and associated documentation files (the "Software"), to deal
 in the Software without restriction, including without limitation the rights
 to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 copies of the Software, and to permit persons to whom the Software is
 furnished to do so, subject to the following conditions:

 The above copyright notice and this permission notice shall be included in all
 copies or substantial portions of the Software.

 THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 SOFTWARE.

 */

import 'jest-styled-components'
import React from 'react'
import { assertSnapshot, renderWithTheme } from '@looker/components-test-utils'
import { fireEvent } from '@testing-library/react'
import { Radio } from './Radio'

test('Radio default', () => {
  assertSnapshot(<Radio id="radioID" />)
})

test('Radio checked set to true', () => {
  assertSnapshot(<Radio checked={true} id="radioID" />)
})

test('Radio checked set to false', () => {
  assertSnapshot(<Radio checked={false} id="radioID" />)
})

test('Radio with name and id', () => {
  assertSnapshot(<Radio id="Chuck" name="Chuck" />)
})

test('Radio should accept disabled', () => {
  assertSnapshot(<Radio disabled id="radioID" />)
})

test('Radio with aria-describedby', () => {
  assertSnapshot(<Radio aria-describedby="some-id" id="radioID" />)
})

test('Radio should trigger onChange handler', () => {
  const onChange = jest.fn()
  const { getByRole } = renderWithTheme(
    <Radio id="radioID" onChange={onChange} />
  )

  const radioInput = getByRole('radio')
  fireEvent.click(radioInput)
  expect(onChange).toHaveBeenCalledTimes(1)
})
