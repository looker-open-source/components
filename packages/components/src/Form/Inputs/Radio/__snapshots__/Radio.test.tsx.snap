// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio checked set to false 1`] = `
.c2 {
  font-family: 'Roboto','Noto Sans JP','Noto Sans CJK KR','Noto Sans Arabic UI','Noto Sans Devanagari UI','Noto Sans Hebrew','Noto Sans Thai UI','Helvetica','Arial',sans-serif;
  background-color: currentColor;
  border: solid 1px #DEE1E5;
  border-color: currentColor;
  border-radius: 50%;
  color: transparent;
  height: 100%;
  position: relative;
  -webkit-transition: background-color 25ms linear,border-color 25ms linear,box-shadow 25ms linear;
  transition: background-color 25ms linear,border-color 25ms linear,box-shadow 25ms linear;
  width: 100%;
}

.c2::after {
  background: #fff;
  border-radius: 50%;
  content: '';
  height: 6px;
  position: absolute;
  right: calc(50% - 3px);
  top: calc(50% - 3px);
  width: 6px;
}

.c0 {
  font-family: 'Roboto','Noto Sans JP','Noto Sans CJK KR','Noto Sans Arabic UI','Noto Sans Devanagari UI','Noto Sans Hebrew','Noto Sans Thai UI','Helvetica','Arial',sans-serif;
  height: 1rem;
  position: relative;
  width: 1rem;
}

.c0 input {
  background: #FFFFFF;
  height: 100%;
  opacity: 0;
  position: absolute;
  width: 100%;
  z-index: 1;
}

.c0 input:checked + .c1 {
  color: #6C43E0;
}

.c0 input:not(:checked) + .c1 {
  background: #FFFFFF;
  border-color: #DEE1E5;
}

.c0 input:focus + .c1 {
  border-color: #9785F2;
  box-shadow: 0 0 0 2px #E8E5FF;
  outline: none;
}

.c0 input:disabled + .c1 {
  color: #939BA5;
}

.c0 input:disabled:not(:checked) + .c1 {
  background: #F5F6F7;
}

.c0 input:disabled:not(:checked) + .c1::after {
  background: #F5F6F7;
}

<div
  className="c0"
>
  <input
    checked={false}
    id="radioID"
    type="radio"
  />
  <div
    className="c1 c2"
  />
</div>
`;

exports[`Radio checked set to true 1`] = `
.c2 {
  font-family: 'Roboto','Noto Sans JP','Noto Sans CJK KR','Noto Sans Arabic UI','Noto Sans Devanagari UI','Noto Sans Hebrew','Noto Sans Thai UI','Helvetica','Arial',sans-serif;
  background-color: currentColor;
  border: solid 1px #DEE1E5;
  border-color: currentColor;
  border-radius: 50%;
  color: transparent;
  height: 100%;
  position: relative;
  -webkit-transition: background-color 25ms linear,border-color 25ms linear,box-shadow 25ms linear;
  transition: background-color 25ms linear,border-color 25ms linear,box-shadow 25ms linear;
  width: 100%;
}

.c2::after {
  background: #fff;
  border-radius: 50%;
  content: '';
  height: 6px;
  position: absolute;
  right: calc(50% - 3px);
  top: calc(50% - 3px);
  width: 6px;
}

.c0 {
  font-family: 'Roboto','Noto Sans JP','Noto Sans CJK KR','Noto Sans Arabic UI','Noto Sans Devanagari UI','Noto Sans Hebrew','Noto Sans Thai UI','Helvetica','Arial',sans-serif;
  height: 1rem;
  position: relative;
  width: 1rem;
}

.c0 input {
  background: #FFFFFF;
  height: 100%;
  opacity: 0;
  position: absolute;
  width: 100%;
  z-index: 1;
}

.c0 input:checked + .c1 {
  color: #6C43E0;
}

.c0 input:not(:checked) + .c1 {
  background: #FFFFFF;
  border-color: #DEE1E5;
}

.c0 input:focus + .c1 {
  border-color: #9785F2;
  box-shadow: 0 0 0 2px #E8E5FF;
  outline: none;
}

.c0 input:disabled + .c1 {
  color: #939BA5;
}

.c0 input:disabled:not(:checked) + .c1 {
  background: #F5F6F7;
}

.c0 input:disabled:not(:checked) + .c1::after {
  background: #F5F6F7;
}

<div
  className="c0"
>
  <input
    checked={true}
    id="radioID"
    type="radio"
  />
  <div
    className="c1 c2"
  />
</div>
`;

exports[`Radio default 1`] = `
.c2 {
  font-family: 'Roboto','Noto Sans JP','Noto Sans CJK KR','Noto Sans Arabic UI','Noto Sans Devanagari UI','Noto Sans Hebrew','Noto Sans Thai UI','Helvetica','Arial',sans-serif;
  background-color: currentColor;
  border: solid 1px #DEE1E5;
  border-color: currentColor;
  border-radius: 50%;
  color: transparent;
  height: 100%;
  position: relative;
  -webkit-transition: background-color 25ms linear,border-color 25ms linear,box-shadow 25ms linear;
  transition: background-color 25ms linear,border-color 25ms linear,box-shadow 25ms linear;
  width: 100%;
}

.c2::after {
  background: #fff;
  border-radius: 50%;
  content: '';
  height: 6px;
  position: absolute;
  right: calc(50% - 3px);
  top: calc(50% - 3px);
  width: 6px;
}

.c0 {
  font-family: 'Roboto','Noto Sans JP','Noto Sans CJK KR','Noto Sans Arabic UI','Noto Sans Devanagari UI','Noto Sans Hebrew','Noto Sans Thai UI','Helvetica','Arial',sans-serif;
  height: 1rem;
  position: relative;
  width: 1rem;
}

.c0 input {
  background: #FFFFFF;
  height: 100%;
  opacity: 0;
  position: absolute;
  width: 100%;
  z-index: 1;
}

.c0 input:checked + .c1 {
  color: #6C43E0;
}

.c0 input:not(:checked) + .c1 {
  background: #FFFFFF;
  border-color: #DEE1E5;
}

.c0 input:focus + .c1 {
  border-color: #9785F2;
  box-shadow: 0 0 0 2px #E8E5FF;
  outline: none;
}

.c0 input:disabled + .c1 {
  color: #939BA5;
}

.c0 input:disabled:not(:checked) + .c1 {
  background: #F5F6F7;
}

.c0 input:disabled:not(:checked) + .c1::after {
  background: #F5F6F7;
}

<div
  className="c0"
>
  <input
    id="radioID"
    type="radio"
  />
  <div
    className="c1 c2"
  />
</div>
`;

exports[`Radio should accept disabled 1`] = `
.c2 {
  font-family: 'Roboto','Noto Sans JP','Noto Sans CJK KR','Noto Sans Arabic UI','Noto Sans Devanagari UI','Noto Sans Hebrew','Noto Sans Thai UI','Helvetica','Arial',sans-serif;
  background-color: currentColor;
  border: solid 1px #DEE1E5;
  border-color: currentColor;
  border-radius: 50%;
  color: transparent;
  height: 100%;
  position: relative;
  -webkit-transition: background-color 25ms linear,border-color 25ms linear,box-shadow 25ms linear;
  transition: background-color 25ms linear,border-color 25ms linear,box-shadow 25ms linear;
  width: 100%;
}

.c2::after {
  background: #fff;
  border-radius: 50%;
  content: '';
  height: 6px;
  position: absolute;
  right: calc(50% - 3px);
  top: calc(50% - 3px);
  width: 6px;
}

.c3 input:checked + .c1 {
  color: #6C43E0;
}

.c3 input:not(:checked) + .c1 {
  background: #FFFFFF;
  border-color: #DEE1E5;
}

.c3 input:focus + .c1 {
  border-color: #9785F2;
  box-shadow: 0 0 0 2px #E8E5FF;
  outline: none;
}

.c3 input:disabled + .c1 {
  color: #939BA5;
}

.c3 input:disabled:not(:checked) + .c1 {
  background: #F5F6F7;
}

.c3 input:disabled:not(:checked) + .c1::after {
  background: #F5F6F7;
}

.c0 {
  font-family: 'Roboto','Noto Sans JP','Noto Sans CJK KR','Noto Sans Arabic UI','Noto Sans Devanagari UI','Noto Sans Hebrew','Noto Sans Thai UI','Helvetica','Arial',sans-serif;
  height: 1rem;
  position: relative;
  width: 1rem;
}

.c0 input {
  background: #FFFFFF;
  cursor: not-allowed;
  height: 100%;
  opacity: 0;
  position: absolute;
  width: 100%;
  z-index: 1;
}

.c0 input:checked + .c1 {
  color: #6C43E0;
}

.c0 input:not(:checked) + .c1 {
  background: #FFFFFF;
  border-color: #DEE1E5;
}

.c0 input:focus + .c1 {
  border-color: #9785F2;
  box-shadow: 0 0 0 2px #E8E5FF;
  outline: none;
}

.c0 input:disabled + .c1 {
  color: #939BA5;
}

.c0 input:disabled:not(:checked) + .c1 {
  background: #F5F6F7;
}

.c0 input:disabled:not(:checked) + .c1::after {
  background: #F5F6F7;
}

<div
  className="c0"
>
  <input
    disabled={true}
    id="radioID"
    type="radio"
  />
  <div
    className="c1 c2"
  />
</div>
`;

exports[`Radio with aria-describedby 1`] = `
.c2 {
  font-family: 'Roboto','Noto Sans JP','Noto Sans CJK KR','Noto Sans Arabic UI','Noto Sans Devanagari UI','Noto Sans Hebrew','Noto Sans Thai UI','Helvetica','Arial',sans-serif;
  background-color: currentColor;
  border: solid 1px #DEE1E5;
  border-color: currentColor;
  border-radius: 50%;
  color: transparent;
  height: 100%;
  position: relative;
  -webkit-transition: background-color 25ms linear,border-color 25ms linear,box-shadow 25ms linear;
  transition: background-color 25ms linear,border-color 25ms linear,box-shadow 25ms linear;
  width: 100%;
}

.c2::after {
  background: #fff;
  border-radius: 50%;
  content: '';
  height: 6px;
  position: absolute;
  right: calc(50% - 3px);
  top: calc(50% - 3px);
  width: 6px;
}

.c0 {
  font-family: 'Roboto','Noto Sans JP','Noto Sans CJK KR','Noto Sans Arabic UI','Noto Sans Devanagari UI','Noto Sans Hebrew','Noto Sans Thai UI','Helvetica','Arial',sans-serif;
  height: 1rem;
  position: relative;
  width: 1rem;
}

.c0 input {
  background: #FFFFFF;
  height: 100%;
  opacity: 0;
  position: absolute;
  width: 100%;
  z-index: 1;
}

.c0 input:checked + .c1 {
  color: #6C43E0;
}

.c0 input:not(:checked) + .c1 {
  background: #FFFFFF;
  border-color: #DEE1E5;
}

.c0 input:focus + .c1 {
  border-color: #9785F2;
  box-shadow: 0 0 0 2px #E8E5FF;
  outline: none;
}

.c0 input:disabled + .c1 {
  color: #939BA5;
}

.c0 input:disabled:not(:checked) + .c1 {
  background: #F5F6F7;
}

.c0 input:disabled:not(:checked) + .c1::after {
  background: #F5F6F7;
}

.c3 input:checked + .c1 {
  color: #6C43E0;
}

.c3 input:not(:checked) + .c1 {
  background: #FFFFFF;
  border-color: #DEE1E5;
}

.c3 input:focus + .c1 {
  border-color: #9785F2;
  box-shadow: 0 0 0 2px #E8E5FF;
  outline: none;
}

.c3 input:disabled + .c1 {
  color: #939BA5;
}

.c3 input:disabled:not(:checked) + .c1 {
  background: #F5F6F7;
}

.c3 input:disabled:not(:checked) + .c1::after {
  background: #F5F6F7;
}

<div
  className="c0"
>
  <input
    aria-describedby="some-id"
    id="radioID"
    type="radio"
  />
  <div
    className="c1 c2"
  />
</div>
`;

exports[`Radio with name and id 1`] = `
.c2 {
  font-family: 'Roboto','Noto Sans JP','Noto Sans CJK KR','Noto Sans Arabic UI','Noto Sans Devanagari UI','Noto Sans Hebrew','Noto Sans Thai UI','Helvetica','Arial',sans-serif;
  background-color: currentColor;
  border: solid 1px #DEE1E5;
  border-color: currentColor;
  border-radius: 50%;
  color: transparent;
  height: 100%;
  position: relative;
  -webkit-transition: background-color 25ms linear,border-color 25ms linear,box-shadow 25ms linear;
  transition: background-color 25ms linear,border-color 25ms linear,box-shadow 25ms linear;
  width: 100%;
}

.c2::after {
  background: #fff;
  border-radius: 50%;
  content: '';
  height: 6px;
  position: absolute;
  right: calc(50% - 3px);
  top: calc(50% - 3px);
  width: 6px;
}

.c0 {
  font-family: 'Roboto','Noto Sans JP','Noto Sans CJK KR','Noto Sans Arabic UI','Noto Sans Devanagari UI','Noto Sans Hebrew','Noto Sans Thai UI','Helvetica','Arial',sans-serif;
  height: 1rem;
  position: relative;
  width: 1rem;
}

.c0 input {
  background: #FFFFFF;
  height: 100%;
  opacity: 0;
  position: absolute;
  width: 100%;
  z-index: 1;
}

.c0 input:checked + .c1 {
  color: #6C43E0;
}

.c0 input:not(:checked) + .c1 {
  background: #FFFFFF;
  border-color: #DEE1E5;
}

.c0 input:focus + .c1 {
  border-color: #9785F2;
  box-shadow: 0 0 0 2px #E8E5FF;
  outline: none;
}

.c0 input:disabled + .c1 {
  color: #939BA5;
}

.c0 input:disabled:not(:checked) + .c1 {
  background: #F5F6F7;
}

.c0 input:disabled:not(:checked) + .c1::after {
  background: #F5F6F7;
}

<div
  className="c0"
>
  <input
    id="Chuck"
    name="Chuck"
    type="radio"
  />
  <div
    className="c1 c2"
  />
</div>
`;
